File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.cpp
class Solution {
public:
    string processStr(string s) {
        string res="";
        for(auto i:s){
            if(islower(i)){
                res+=i;
            }
            if(i=='#'){
                res+=res;
            }
            if(i=='*' && !res.empty()){
                res.pop_back();
            }
            if(i=='%'){
                reverse(res.begin(),res.end());
            }
        }
        return res;
    }
};

File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.py
class Solution:
    def processStr(self, s: str) -> str:
        st=""
        for i in s:
            if i>='a' and i<='z':
                st+=i
            if i=='#' and len(i)>0:
                st+=st
            if i=='*':
                st=st[:-1]
            if i=='%':
                st=st[::-1]
        return st